crates/store/src/state.rs

```diff
@@ -653,25 +660,83 @@ impl State {
 
     /// Lists all known nullifiers with their inclusion blocks, intended for testing.
     pub async fn list_nullifiers(&self) -> Result<Vec<(Nullifier, u32)>, DatabaseError> {
-        self.db.select_nullifiers().await
+        self.db.select_all_nullifiers().await
     }
 
     /// Lists all known accounts, with their ids, latest state hash, and block at which the account
     /// was last modified, intended for testing.
     pub async fn list_accounts(&self) -> Result<Vec<AccountInfo>, DatabaseError> {
-        self.db.select_accounts().await
+        self.db.select_all_accounts().await
     }
 
     /// Lists all known notes, intended for testing.
     pub async fn list_notes(&self) -> Result<Vec<NoteRecord>, DatabaseError> {
-        self.db.select_notes().await
+        self.db.select_all_notes().await
     }
 
-    /// Returns details for public (public) account.
+    /// Returns details for public (on-chain) account.
     pub async fn get_account_details(&self, id: AccountId) -> Result<AccountInfo, DatabaseError> {
         self.db.select_account(id).await
     }
 
+    /// Returns account states with details for public accounts.
+    pub async fn get_account_states(
+        &self,
+        account_ids: HashSet<AccountId>,
+        include_headers: bool,
+    ) -> Result<(BlockNumber, Vec<AccountProofsResponse>), DatabaseError> {
+        // Lock inner state for the whole operation. We need to hold this lock to prevent the
+        // database, account tree and latest block number from changing during the operation,
+        // because changing one of them would lead to inconsistent state.
+        let inner_state = self.inner.read().await;
+
+        let state_headers = if !include_headers {
+            HashMap::<AccountId, AccountStateHeader>::default()
```

Comment:
Is there a reason to use `HashMap` rather than `BTreeMap` here? I don't really have a strong preference, but we usually use `BTreeMpas` everywhere.

Reply:
I also don't have strong preference here. Taking that this map will be pretty small, it shouldn't be any significant difference between them, so I will use `BTreeMap` here, thanks!